src/scene_graph/mesh_node.hpp

```diff
@@ -8,10 +8,109 @@ namespace wr
 	struct MeshNode : Node
 	{
 		Model* m_model;
+		std::vector<MaterialHandle> m_materials;
 
 		DirectX::XMVECTOR m_aabb[2];
 
-		MeshNode(Model* model) : m_model(model) { }
+		MeshNode(Model* model) : m_model(model) 
```

Comment:
Please add a source file for all of the implementations of MeshNode. So you don't have to include any files and can use predeclaration.